R/protein_abundance_normalisation.R

```diff
@@ -0,0 +1,262 @@
+#' are required. Ideally the output of \code{assign_missingness} or \code{impute} is used for the method \code{"ref_vs_rest"}, since here also the variable 
+#' \code{missingness} is required in contrast to method \code{"all_vs_all"}.
+#' @param tc_proteins A data frame containing tryptic control protein level data. The \code{sample, protein_id, condition, protein_intensity} variables
+#' are required. Ideally the output of \code{assign_missingness} or \code{impute} is used for the method \code{"ref_vs_rest"}, since here also the variable 
+#' \code{missingness} is required in contrast to method \code{"all_vs_all"}.
+#' @param sample The column in the data frame containing the sample name. Is not required if \code{method = "t-test_mean_sd"}.
+#' @param protein_id The column in the data frame containing the protein accession numbers. 
+#' @param grouping The column in the data frame containing precursor or peptide identifiers.
+#' @param condition The column in the data frame containing the conditions.
+#' @param peptide_intensity The column in the data frame containing peptide intensity values. The intensity values need to be log2 transformed.
+#' @param protein_intensity The column in the data frame containing protein intensity values. The intensity values need to be log2 transformed.
+#' @param method A character vector specifying which method should be used. If \code{method = "all_vs_all"} an ANOVA based method will be performed that 
+#' compares all conditions with each other followed by Tukey's honestly significant difference test to identify pairs of conditions that are significantly different.
+#' If \code{mehtod = "ref_vs_rest"} a reference condition will be compared against all other conditions using a Welch t-test in order to identify significantly
+#' changing peptides or precursors.
+#' @param paired A logical indicating if LiP and tryptic control data are paired. In other words it indicates if replicates of LiP and tryptic control belong together.
+#' @param replicate_index The column in the data frame containing numbering for condition replicates. This column is only required if LiP and tryptic control samples are paired.
+#' In that case the corresponding samples in both files should have the same numbering.
+#' @param comparison The column in the data frame containing comparison information of treatment/reference condition pairs. Can be obtained by 
+#' calling \code{assign_missingness}. 
+#' @param completeness The minimal degree of data completeness to be considered for evaluation. Value has to be between 0 and 1, default is 0.7. It is multiplied 
+#' with the number of replicates and then adjusted downward. The resulting number is the minimal number of observations for each condition to be considered for
+#' evaluation.
+#'
+#' @return A data frame that contains significance test statistics and differential abundances (\code{diff}) for each peptide or precursor and the 
```

Comment:
I am not sure, if I understand this correctly. Are the values normalized or do you calculate an ANOVA and/or a t-test? The output is the test statistics and differences in abundances, not normalized values, is that correct? 
Then the function name might be misleading.

Reply:
The output is normalised differences in abundances and the test statistics for these.

Reply:
The output is still a ratio, not a normalised intensity though, right? I find this a bit misleading

Reply:
What I mean is that if the data was normalised, the peptide intensities would be same if "only" the protein abundance changes.